spec.bs

```diff
@@ -134,33 +192,99 @@ Modify {{Document/requestStorageAccess()}} at step 14.1.1.1.1 to read:
 
 <h3 id="storage">Changes to various client-side storage mechanisms</h3>
 
-<h4 id="dom-storage">DOM Storage</h4>
+For all of the following getters and methods, consider the |docForSAA| to be |doc| with the following modifications:
 
-TBD
+1. When attempting to [=obtain a storage key=] the returned key will use [[STORAGE-PARTITIONING#relaxing-additional-keying]] if the tuple does not simply contain an [=/origin=].
 
-<h4 id="indexed-db">IndexedDB</h4>
+Issue(19): Clarify client-side storage mechanism changes in more detail.
 
-TBD
+<h4 id="dom-storage">[=Web storage=]</h4>
 
-<h4 id="web-locks">Web Locks</h4>
+When invoked on {{StorageAccessHandle}} |handle| with {{StorageAccessTypes}} |types| and {{Document}} |docForSAA|, the <dfn export attribute for=StorageAccessHandle><code>sessionStorage</code></dfn> getter must run these steps:
```

Comment:
I think we need to make it clear how the handle and types are connected, we can't just conjure the types out of nowhere. Maybe, below the definition of StorageAccessHandle, add roughly something like this:

```
A {{StorageAccessHandle}} object has an associated {{StorageAccessTypes}} <dfn for=StorageAccessHandle>types</dfn>.
```

then when creating the handle you could say
```
Set |handle|'s [=StorageAccessHandle/types=] to |types|.
```

and take the types from the handle when using it here and in the instances below.